### PR TITLE
UI-493-Button Group

### PR DESCRIPTION
**Pull request by sheela-svmx**
Button Group

### CODE REVIEW

File: lib/components/Button/ButtonGroup.js [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
+  children: PropTypes.node,
+  /**
+   * className that will be used for styling the wrapper div and its children for ButtonGroup
+   */
+  className: PropTypes.string,
+};
 
-  renderButton(button, index) {
-    const cnt = React.Children.count(this.props.children);
-    if (button.type && (button.type === DropdownButton || button.type.isGroupable)) {
```
Comment (sheela-svmx) [inline lib/components/Button/ButtonGroup.js]: This logic should not be tied to Button group, should be a part of Dropdown

**Comment by sheela-svmx** [comment]
@joshsylvester @hanzge @mallik-svmx @kokasvmx

File: lib/components/Button/ButtonGroup.js [hunk 2]
```diff
@@ -1,43 +1,38 @@
...
-      return React.cloneElement(button, {
-        key: index,
-        grouped: true,
-        isFirstInGroup: index === 0,
-        isLastInGroup: index === cnt - 1,
-      });
-    }
+const defaultProps = {
+  children: [],
+  className: '',
```
Comment (maxsonar) [inline lib/components/Button/ButtonGroup.js]: BLOCKER Remove this trailing comma. [rule](https://sonarqube.servicemax.io/coding_rules#rule_key=javascript%3ATrailingComma)

File: lib/components/Button/ButtonGroup.js [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
+};
 
-    return button;
-  }
+const ButtonGroup = ({ ...props }) => {
```
Comment (maxsonar) [inline lib/components/Button/ButtonGroup.js]: MINOR Rename this 'ButtonGroup' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](https://sonarqube.servicemax.io/coding_rules#rule_key=javascript%3AS100)

File: lib/components/Button/ButtonGroup.js [hunk 4]
```diff
@@ -1,43 +1,38 @@
...
-  }
+const propTypes = {
+  /**
+   * Children components for the ButtonGroup
+   */
+  children: PropTypes.node,
+  /**
+   * className that will be used for styling the wrapper div and its children for ButtonGroup
+   */
+  className: PropTypes.string,
```
Comment (maxsonar) [inline lib/components/Button/ButtonGroup.js]: BLOCKER Remove this trailing comma. [rule](https://sonarqube.servicemax.io/coding_rules#rule_key=javascript%3ATrailingComma)

**Comment by maxsonar** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 2 blocker
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by maxio** [comment]
Test FAILed.
Refer to this link for build results: https://jenkins.servicemax.io/job/UIComponents/job/ui-components-lightning/8/.

File: lib/components/Button/ButtonGroup.js (hunk 3 above)
Comment (mallik-svmx) [inline lib/components/Button/ButtonGroup.js]: Just pass the props here, as you are de-structuring in next line# 22

File: lib/components/Button/ButtonGroup.js [hunk 5]
```diff
@@ -1,43 +1,38 @@
...
-      });
-    }
+const defaultProps = {
+  children: [],
+  className: '',
+};
 
-    return button;
-  }
+const ButtonGroup = (props) => {
```
Comment (maxsonar) [inline lib/components/Button/ButtonGroup.js]: MINOR Rename this 'ButtonGroup' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](https://sonarqube.servicemax.io/coding_rules#rule_key=javascript%3AS100)

**Comment by maxio** [comment]
Test FAILed.
Refer to this link for build results: https://jenkins.servicemax.io/job/UIComponents/job/ui-components-lightning/10/.

**Comment by sheela-svmx** [comment]
@mallik-svmx updated

File: lib/components/Button/ButtonGroup.js [hunk 6]
```diff
@@ -1,43 +1,38 @@
...
+  const { className, children, ...pprops } = props;
```
Comment (joshsylvester) [inline lib/components/Button/ButtonGroup.js]: please rename the `pprops` to something else. `groupProps` or something

Comment (mallik-svmx) [reply lib/components/Button/ButtonGroup.js]: ...rest Or ...restProps ?

**Comment by sheela-svmx** [comment]
@joshsylvester updated

**Pull request merged**